operations/network/dns/PowerDNS_Troubleshooting.md

```diff
@@ -0,0 +1,141 @@
+│   │   └── uai_macvlan_bridge.nmn.wasp.dev.cray.com.
+...
+```
+
+For more information on External DNS and troubleshooting steps see the [External DNS documentation](../external_dns/External_DNS.md)
+
+## PowerDNS logging
+
+When troubleshooting DNS problems it may prove helpful to increase the level of logging from the default value of 3 (error).
+
+1. Edit the cray-dns-powerdns ConfigMap
+
+   ```
+   ncn-m001:~ # kubectl -n services edit cm cray-dns-powerdns
+   ```
+
+1. Set the `loglevel` parameter in `pdns.conf` to the desired setting
+
+   ```
+   pdns.conf: |
+      config-dir=/etc/pdns
+      include-dir=/etc/pdns/conf.d
+      guardian=yes
+      loglevel=3
+      setgid=pdns
+      setuid=pdns
+      socket-dir=/var/run
+      version-string=anonymous
+   ```
+
+1. Restart the PowerDNS service
+
+   ```
+   ncn-m001:~ # kubectl -n services rollout restart deployment cray-dns-powerdns
+   deployment.apps/cray-dns-powerdns restarted
+   ```
+
+Please see the [PowerDNS documentation](https://doc.powerdns.com/authoritative/settings.html#loglevel) for more information.
+
+## Verify DNSSEC operation
+
+### Verify zones are being signed with the zone signing key
+
+Check that the required zone has a DNSKEY entry, this should match the public key portion of the zone signing key.
+
+```
+ncn-m001:~ # kubectl -n services exec deployment/cray-dns-powerdns -c cray-dns-powerdns -- pdnsutil show-zone wasp.dev.cray.com
+This is a Master zone
+Last SOA serial number we notified: 2021090901 == 2021090901 (serial in the database)
+Zone has following allowed TSIG key(s): wasp-key
+Zone uses following TSIG key(s): wasp-key
+Metadata items:
+	AXFR-MASTER-TSIG	wasp-key
+	SOA-EDIT-API	DEFAULT
+	TSIG-ALLOW-AXFR	wasp-key
+Zone has NSEC semantics
+keys:
+ID = 1 (CSK), flags = 257, tag = 26690, algo = 13, bits = 256	  Active	 Published  ( ECDSAP256SHA256 )
+CSK DNSKEY = wasp.dev.cray.com. IN DNSKEY 257 3 13 TAi+aXL+Z8ZSFHxz+iEWB3MEdi1JWgM/tb3Q1M76yVOq5Kaur9k+oIAHXvCSR19Iuu+0ZUAyLB0vKkhScJp3Tw== ; ( ECDSAP256SHA256 )
+DS = wasp.dev.cray.com. IN DS 26690 13 1 8c926281afb822a2bea767f08c79b856a2427c26 ; ( SHA1 digest )
+DS = wasp.dev.cray.com. IN DS 26690 13 2 2bfd71e5403f99d25496f5f7f352e71747bb72ee6eb240dcaf8b56b95d18ef6c ; ( SHA256 digest )
+DS = wasp.dev.cray.com. IN DS 26690 13 4 df40f23a7ee051d7e3d40d4059640bda3558cd74a37110b25f7b8cf4e60506c77bf33a660400710d397df0a1cde26d70 ; ( SHA-384 digest )
+```
+If the DNSKEY record is incorrect verify that the zone name is correct in the `dnssec` SealedSecret in `customizations.yaml` and that the desired zone signing key was used. Please see the [PowerDNS Configuration Guide](./PowerDNS_Configuration.md) for more information.
+
+### Verify TSIG operation
+
+> **`IMPORTANT`** these examples are for informational purposes only. The use of the `dig` -y option to present the key should be avoided in favour of the -k option with the secret in a file to avoid the key being displayed in `ps` output or the shell history.
```

Comment:
Change to:

> **IMPORTANT:** These examples are for informational purposes only. The use of the `dig` command `-y` option to present the key should be avoided in favor of the `-k` option with the secret in a file to avoid the key being displayed in `ps` output or the shell history.